src/ObjCBindings/FieldAttribute.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+namespace ObjCBindings {
+
+	/// <summary>
+	/// Generic attribute that is used to mark code to be backed by a field in ObjC.
+	/// </summary>
+	[Experimental ("APL0003")]
+	[AttributeUsage (AttributeTargets.Property | AttributeTargets.Field)]
+	public sealed class FieldAttribute<T> : Attribute where T : FieldTag {
```

Comment:
Why isn't the existing FieldAttribute good enough?

Reply:
It can be good enough, but it makes the implementation of the specific analyser are complicated. Adding the tags makes it very simple to let an analyser decide what fields to pay attention to or not. For example, if we have an analyser that just looks at Notifications, it can be more efficient at runtime to filter for the specific generic implementation.